### PR TITLE
Refactor reuse some simple db.SearchOrderBy

### LINKED ISSUE #25808: Using db.SearchOrderBy in internal instead of sortType
### Feature Description

As we have a globally `db.SearchOrderBy` for sorting, in some places we still using `xorm.Session.Asc(col_name)` and `xorm.Session.Desc(col_name)` which may make mistakes like #25806 and hard to maintain.

I have made a simple keyword search, and found there are two places is using `Asc` and `Desc` directly:
- models/git/commit_status.go: func sortCommitStatusesSession
- models/issues/issue_search.go: func applySorts

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by CaiCandong**
As @yp05327 suggested in #25808  , we should use `db.SearchOrderBy` instead of `sortType` whenever possible. Considering that `db.SearchOrderBy` can't cover all the complex sorting logic possible, I've made a simple modification to the implementation in improve in #25806.
related #25806
close #25808

### CODE REVIEW

**Comment by wxiaoguang** [comment]
I do not think it right to add more "db.SearchOrderXxx", it's a over-design.

In the end, every model/table should have their own "order rules", but not using the shared unclear one.

**Comment by wxiaoguang** [comment]
The  over-designed/over-abstracted shared constants / shared templates already causes problems like #14947

**Comment by CaiCandong** [comment]
> I do not think it right to add more "db.SearchOrderXxx", it's an over-design/over-abstraction.
> 
> In the end, every model/table should have its own "order rules", but not use the shared unclear one.

I also think that having the same "order rules" common to every model/table is a bit of over-design, so I somehow backed out the `improve` in #25806 in my implementation. 

While keeping each model/table independent, reusing some simple sorting rules can also reduce the possibility of errors. I think there is a trade-off between reuse and independence. 

Maybe I should change the title of this PR.

**Comment by yp05327** [comment]
Maybe we can design a global shareable order rules, like the usage of `db.SearchOrderByXXX` now.
And for each model/table they have their own "order rules", and some of these rules can reference the globally shared order rules.

File: models/git/commit_status.go [hunk 1]
```diff
@@ -254,17 +254,17 @@ func listCommitStatusesStatement(ctx context.Context, repo *repo_model.Repositor
 func sortCommitStatusesSession(sess *xorm.Session, sortType string) {
 	switch sortType {
 	case "oldest":
-		sess.Asc("created_unix")
+		sess.OrderBy(db.SearchOrderByOldest.String())
```
Comment (lunny) [inline models/git/commit_status.go:257]: How about rewrite this function like the previous `GetSearchOrderByBySortType` implementation?

Comment (CaiCandong) [reply models/git/commit_status.go:257]: I think the sorting rules are independent for each model/table, the way I wrote it I just want to reuse some common simple sorting rules. If we use `GetSearchOrderByBySortType`, then each model needs to write a `GetSearchOrderByBySortType`, and then `db.SearchOrderBy` needs to contain all the sorting rules, which is overdesigned.
While sortType is of type `string`, we can sortType to choose to reuse `db.SearchOrderBy` or define our own sorting rules. For example:

```
switch sortType {
	case "oldest":
		sess.OrderBy(db.SearchOrderByOldest.String())
       case "nearduedate":
               sess.Join("LEFT", "milestone", "issue.milestone_id = milestone.id").
			OrderBy("CASE " +
				"WHEN issue.deadline_unix = 0 AND (milestone.deadline_unix = 0 OR milestone.deadline_unix IS NULL) THEN 253370764800 " +
				"WHEN milestone.deadline_unix = 0 OR milestone.deadline_unix IS NULL THEN issue.deadline_unix " +
				"WHEN milestone.deadline_unix < issue.deadline_unix OR issue.deadline_unix = 0 THEN milestone.deadline_unix " +
				"ELSE issue.deadline_unix END ASC").
			Desc("issue.created_unix").
			Desc("issue.id")
```

**Pull request closed without merging**